src/java/org/apache/cassandra/index/sai/disk/v1/V1OnDiskFormat.java

```diff
@@ -222,27 +222,27 @@ public boolean validatePerIndexComponents(IndexDescriptor indexDescriptor, Index
     {
         for (IndexComponent indexComponent : perIndexComponents(indexContext))
         {
-            // VSTODO: lucene doesn't follow SAI naming patterns and manage its own validation
-            if (!isBuildCompletionMarker(indexComponent) && !(indexContext.isVector()))
+            if (isBuildCompletionMarker(indexComponent))
```

Comment:
What happens if we try to read a file that was written by code that didn't include the checksum?